sql/core/src/main/scala/org/apache/spark/sql/execution/HiveResult.scala

```diff
@@ -67,12 +67,38 @@ object HiveResult {
       command.executeCollect().map(_.getString(1))
     case other =>
       val timeFormatters = getTimeFormatters
-      val result: Seq[Seq[Any]] = other.executeCollectPublic().map(_.toSeq).toSeq
-      // We need the types so we can output struct field names
-      val types = executedPlan.output.map(_.dataType)
-      // Reformat to match hive tab delimited output.
-      result.map(_.zip(types).map(e => toHiveString(e, false, timeFormatters)))
-        .map(_.mkString("\t"))
+      val commandPhysicalPlans = other collect {
```

Comment:
I think it's safer to only handle root command here, as that's what Hive can support.

Reply:
Basically we just add more cases above:
```
case command @ CommandResultExec(_,  _: ShowTablesExec, _) => ...
case ...
```

Reply:
Basically, we just strip the root `CommandResultExec` at the beginning
```
... = stripRootCommandResult(executedPlan) match {
  ...
}
```

Reply:
OK